modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -757,7 +757,9 @@ public void mergeGuestCommerceOrder(
 				for (CommerceOrderItem userCommerceOrderItem :
 						userCommerceOrderItems) {
 
-					if (_jsonHelper.equals(
+					if ((Validator.isNull(guestCommerceOrderItem.getJson()) &&
+						 Validator.isNull(userCommerceOrderItem.getJson())) ||
+						_jsonHelper.equals(
```

Comment:
what happens if only one of the two (guestCommerceOrderItem.getJson(), userCommerceOrderItem.getJson())  is null?

Reply:
Hmmm good point. I haven't seen that case but it seems like we should handle null checking in the jsonHelper anyway. Comparing 2 strings where both are null should be true and comparing 2 strings where one is null and the other is not should be false.

Reply:
If we left it as is, and one of the jsons was null it would continue to throw an error from the jsonHelper.

Reply:
@alecsloan you didn't push commit for this third case, right?